backend/consensus/base.py

```diff
@@ -97,9 +100,10 @@ async def exec_transaction(
 
         print(" ~ ~ ~ ~ ~ EXECUTING TRANSACTION: ", transaction)
         # Update transaction status
-        self.transactions_processor.update_transaction_status(
-            transaction["id"], TransactionStatus.PROPOSING.value
-        )
+        with self.dbclient.get_session() as session:
```

Comment:
@AgustinRamiroDiaz why do we need to pass the session to the `update_transaction_status` method instead of the TransactionsProcessor using the session it created in the __init__?

Reply:
Since the `TransactionsProcessor` is only created once at the init of `ConsensusAlgorithm` class, the session will not be usable
The session that I have injected in the top `server.py` module is a session automatically created by `Flask-SQLAlchemy` that is bound to each request in the Flask server.
This is related to the problem I'm mentioning in the description around the refactor and how to inject sessions to cronjobs

Reply:
changed and now it's injected :+1: